jsonrpc-specs/src/methods/discv5.json

```diff
@@ -0,0 +1,567 @@
+				"required": [
+					"enr",
+					"nodeId"
+				],
+				"properties": {
+					"enr": {
+						"title": "nodeENR",
+						"description": "URL-safe base64 encoded \\\"text\\\" version of the ENR prefixed by \\\"enr:\\\".\"",
+						"$ref": "#/components/schemas/Enr"
+					},
+					"nodeId": {
+						"title": "nodeId",
+						"description": "Hex encoded `NodeId` of an ENR (a 32 byte identifier).",
+						"$ref": "#/components/schemas/bytes32"
+					}
+				}
+			}
+		}
+	},
+	{
+		"name": "discv5_routingTableInfo",
+		"summary": "Returns meta information about discv5 routing table.",
+		"params": [],
+		"result": {
+			"name": "routingTableInfoResult",
+			"description": "Discv5 routing table information",
+			"required": true,
+			"schema": {
+				"title": "routingTableInfoResults",
+				"description": "Routing table details",
+				"type": "object",
+				"required": [
+					"localKey",
+					"buckets"
+				],
+				"properties": {
+					"localKey": {
+						"title": "nodeId",
+						"description": "The key identifying the local peer that owns the routing table.",
+						"$ref": "#/components/schemas/bytes32"
+					},
+					"buckets":  {
+						"title": "kBucketsTable",
+						"description": "Represents a Kademlia routing table.",
+						"$ref": "#/components/schemas/kBuckets"
+					}
+				}
+			}
+		}
+	},
+	{
+		"name": "discv5_setEnr",
+		"summary": "Write an ethereum node record to the database.",
+		"params": [
+			{
+				"name": "enr",
+				"required": true,
+				"schema": {
+					"title": "Ethereum node record",
+					"$ref": "#/components/schemas/Enr"
+				}
+			}
+		],
+		"result": {
+			"name": "setEnrResult",
+			"description": "Returns ethereum node record if the node record has been successfully saved.",
+			"schema": {
+				"title": "Ethereum node record",
+				"$ref": "#/components/schemas/Enr"
+			}
+		}
+	},
+	{
+		"name": "discv5_getEnr",
+		"summary": "Fetch the latest ENR associated with the given node ID",
+		"params": [
+			{
+				"name": "node_id",
+				"required": true,
+				"schema": {
+					"title": "NodeId",
+					"$ref": "#/components/schemas/bytes32"
+				}
+			}
+		],
+		"result": {
+			"name": "getEnrResult",
+			"description": "Returns latest ENR associated with the given node ID.",
+			"schema": {
+					"title": "Ethereum node record",
+					"$ref": "#/components/schemas/Enr"
+				      }
+		}
+	},
+	{
+		"name": "discv5_deleteEnr",
+		"summary": "Delete a Node ID from the local database",
+		"params": [
+			{
+				"name": "node_id",
+				"required": true,
+				"schema": {
+					"title": "NodeId",
+					"$ref": "#/components/schemas/bytes32"
+				}
+			}
+		],
+		"result": {
+			"name": "deleteEnrResult",
+			"description": "Returns None upon successful deletion of the node record.",
+			"schema": {
+				"type": "string",
+				"pattern": "None$"
+			}
+		}
+	},
+	{
+		"name": "discv5_lookupEnr",
+		"summary": "Fetch the ENR representation associated with the given Node ID and optional sequence number",
```

Comment:
What is the use case of the sequence number here?

Also, what is actually the difference with the `discv5_recursiveFindNodes` call?

Reply:
> Also, what is actually the difference with the discv5_recursiveFindNodes call?

To answer my own question, the latter should return closest nodes (max 16?) while the first should return the specific ENR that is being looked after.

Reply:
>To answer my own question, the latter should return closest nodes (max 16?) while the first should return the specific ENR that is being looked after.

It seems so and the sequence number is ENR `seq` . I think changing this parameter from `sequence_number` to `enrSeq` is more clear.

Reply:
I think I see what is going on here:

I believe these methods were originally intended to operate on the ENR database itself.  I think the intent was:

- `addENR`: write an ENR record to the ENR database (no network activity)
- `getENR`: read an ENR from the ENR database
- `deleteENR`: delete an ENR from the ENR database
- `lookupENR`: retrieve the ENR record with a sequence number `>=` the one provided.  If we have one in our local database that satisfies `enr.sequence_number >= sequence_number`, return that, otherwise, go get the latest version from the network and return it (even if the sequence number is lower).